packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -212,8 +214,14 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
         )
       }
 
+      const isLegacyIdentifier = credentialOffer.cred_def_id.match(legacyIndyCredentialDefinitionIdRegex)
+      if (!isLegacyIdentifier && useLegacyProverDid) {
+        throw new AriesFrameworkError('Cannot use legacy prover_did with non-legacy identifiers')
+      }
+
       createReturnObj = CredentialRequest.create({
-        entropy: anoncreds.generateNonce(), // FIXME: find a better source of entropy
+        entropy: !useLegacyProverDid || !isLegacyIdentifier ? anoncreds.generateNonce() : undefined, // FIXME: find a better source of entropy
+        proverDid: useLegacyProverDid ? generateLegacyProverDidLikeString() : undefined,
```

Comment:
As it should be random, do wr need to make this a secure ranom generation? E.g. nonce and then encode it as base58?

We could also add a getRandomBytes method to the wallet?

Reply:
I copied it from anoncreds/indy-sdk package implementation, but also thought that could be a good idea to rely in anoncreds native library instead of uuid as it's being done currently in `generateLegacyProverDidLikeString`.

Any 22-bytes long base58 string will be good for that purpose, right?

Reply:
As it's used for entropy, it may be needed for the string to be generated with actual secure random bytes instead of a random string?